ds13/code/modules/mob/living/psychosis.dm

```diff
@@ -0,0 +1,182 @@
+	cure_message = "Life doesn't seem quite as absurd as it used to."
+
+/datum/medical_effect/irrational_laughter/on_life(mob/living/carbon/human/H, strength)
+	var/obj/item/organ/external/head/head = H.get_organ("head")
+	if(istype(head))
+		switch(strength)
+			if(1 to 10)
+				H.custom_pain("You suddenly feel an urge to laugh.",0, affecting = LOWER_TORSO)
+			if(11 to 30)
+				H.custom_pain("You burst out laughing!",1, affecting = LOWER_TORSO)
+			if(31 to INFINITY)
+				H.custom_pain("You burst out into an uncontrollable fit of laughter!",1, affecting = LOWER_TORSO)
+
+/datum/medical_effect/irrational_laughter/manifest(mob/living/carbon/human/H) //Override to move away from a reagent based trigger to an event based one.
+	for(var/R in cures) //If they have a cure inside of them, fail to manifest.
+		if(!H.reagents || H.reagents.has_reagent(R))
+			return FALSE
+	return TRUE
+
+// =======
+
+/**
+*
+*
+* A proc which handles fluff FX for people going mad.
+* @Param stage : How severe the psychosis is. This is a scaling value based on the mob's current psychosis damage (which is increased via witnessing traumatic events, or being touched by necros).
+*
+*/
+
+/mob/living/carbon/human/proc/set_psychosis(var/stage, var/override)
+	if(!can_set_psychosis && !override) //Avoids patients being permanently debilitated by psychosis such that they can't move. EG: if youre on 0 sanity, you have a small window to get help before the effects manifest again.
+		return
+	switch(stage)
+		if(0)
+			clear_fullscreen("insane")
+		if(1)
+			add_side_effect("Irrational laughter")
+		if(2)
+			adjust_hallucination(10,10)
+			to_chat(src, "<span class='warning'>You feel unhinged.</span>")
+			overlay_fullscreen("insane", /obj/screen/fullscreen/insane)
+		if(3 to INFINITY)
+			adjust_hallucination(20,20)
+			to_chat(src, "<span class='warning'>You can see the walls dividing reality fall apart at your feet...</span>")
+			overlay_fullscreen("insane", /obj/screen/fullscreen/insane)
+	can_set_psychosis = FALSE
+	addtimer(CALLBACK(src, .proc/allow_psychosis), 10 SECONDS) ///This cooldown should be balanced around how debilitating psychosis damage actually turns out to be.
+
+/mob/living/carbon/human/proc/allow_psychosis()
+	can_set_psychosis = TRUE
+
+/mob/living/carbon/human/proc/allow_psychosis_message()
+	sanity_cooldown = FALSE
+
+/obj/screen/fullscreen/insane
+	icon_state = "insane"
+	layer = DAMAGE_LAYER
+	alpha = 180
+
+/**
+*
+*
+* A series of procs to handle psychosis damage on mobs.
+*
+*/
+
+/mob/living/carbon/human
+	var/sanity = 100 //Flat value to represent how ""sane"" someone is. Losing sanity is not recommended.
+	var/maxSanity = 100
+	var/last_sanity = 100
+	var/can_set_psychosis = TRUE //This variable means that mobs can't instant descend into total insanity just by getting sneezed on by a necromorph.
+	var/psychosis_immune = FALSE //SET THIS TO TRUE FOR NECROMORPHS IF YOU MAKE THEM A HUMAN SUBTYPE, OR THEY WILL GO INSANE FROM MARKER STUFF
```

Comment:
Rather than a flat value, i recommend making this a multiplier, so partial resistance can be a thing too